### ISSUE TITLE
Yle Areena can not play all content

### ISSUE THREAD

**Issue by jussike** [szymex/xbmc-finnish-tv#72]
Some videos are playable and some not.
Error message: "One or more items failed to play. Check the log file for details."

Here is my kodi.log. It contains one success and one fail. My device is Raspberry Pi with latest Raspbmc build.
https://db.tt/UWeHEKFh

**Comment by szymex** [comment]
Can you provide video title that did not work?

**Comment by jussike** [comment]
It's not difficult to find some video title which fails.
In my log file, I used this path: Areena -> Programs -> Afrikan jako -> Afrikan jako

Similar failing item for example: Areena -> Programs -> 10 kirjaa rikoksesta -> 10 kirjaa rikoksesta episode 10.

In my log, successfully played title was Kaukasia 30 päivässä, episode 3.
Issue is consistent between tries.
Probably the plugin uses too old version of yle-dl library...?

**Comment by szymex** [comment]
I could not reproduce your problem. I was able to play both videos with Kodi on Windows and Raspberry PI.

**Comment by jussike** [comment]
That's little weird.
These videos can be played on OS X too. Then I found title "Alamaailma" which stops every time after one second on OS X but works well on Raspberry Pi.
I could try to debug that problem on Raspberry Pi when I have some time...
Which distro and kodi version you have on Pi?

**Comment by szymex** [comment]
I use latest OpenELEC 5.0.0.

**Comment by hemuli** [comment]
Having the same yle areena issue with  raspberry pi and  latest raspbmc build running kodi (24 Dec 2014). Only programs marked as "viewable abroad" work. No similar issues with openelec though, but cannot use it for other hw compatibility issues.

**Comment by szymex** [comment]
Are you using it from abroad?

**Comment by hemuli** [comment]
Nope, I'm in Finland for sure. Also my dsl provider is one of the biggest ones. There was an update to raspbmc and kodi last night but no change to this. It's consistently refusing to play all areena contents which are for domestic use only. Strange.

**Comment by szymex** [comment]
That's weird. Are you able to play yle-areena videos from same computer through browser? Could there be some proxy configured?

**Comment by hemuli** [comment]
It's pretty vanilla raspbmc install on raspberry pi, so no browser option unfortunately. Most settings are default values. On same hw latest openelec works fine. But cannot use it as primary system due to hw compatibility issues. 

Could there be differences in python-crypto versions (I used apt-get to install it). It's just about the only additional bit I've installed.

**Comment by hemuli** [comment]
I may have found a fix to the issue - changed line 75 in yleareena.py to "useSocks = int(settings.getSetting('socks-use'))" and now can stream all areena content without problems.

**Comment by jussike** [comment]
This really fixes the problem, thank you :)
Python thinks that "0" > 0.

But casting to int is better to do after this if-block
if useSocks == 'no':

**Comment by hemuli** [comment]
Fair point, this was quick 'n dirty :-)

**Comment by anttitelio** [comment]
Err, could you elaborate that, please? I can't get the fix working. The plugin works just as bad as before the change.

**Comment by jussike** [comment]
I'm using this simple patch.
```
--- yleareena.py.original	2015-02-15 22:51:50.221594305 +0200
+++ yleareena.py.patched	2015-02-15 22:53:00.298316393 +0200
@@ -76,6 +76,7 @@
 	if useSocks == 'no':
 		useSocks = 0
 
+	useSocks = int(useSocks)
 	if useSocks > 0 and ( (clip != None and clip['international'] == False) or useSocks == 1):
 		rtmpUrl += " socks=%s" % settings.getSetting('socks-server')
 
```

**Comment by nuija1977** [comment]
For me patch above does not fix anything. Some videos work, some don't. For example all episodes of "Erikoisosasto Stuttgart" work, but none of the episodes of "Erämaan armoilla" work. No proxy is used. Videos work fine when using android tablet. (Same network)

Setup:
Raspberry Pi + Raspbmc
Yle areena plugin 1.7.2

**Comment by anttitelio** [comment]
I get the same results with those episodes and the patch.

RPi model B, OpenElec 5.0.3, Yle Areena plugin 1.7.2, Outside Finland = no, Use a SOCKS proxy = no.

**Comment by amergin** [comment]
Another example of content that cannot be played: [Outlander - Matkantekijä (16) #13](http://areena.yle.fi/1-2354824) (Script failed).

**Comment by bzar** [comment]
Ran into this today with the latest Uutisvuoto. Looking at the log it tried to use localhost:1080 as SOCKS proxy for some videos. Fixed like others have here, by casting useSocks to int before use.
